pkg/standalone-utils/contracts/ProtocolFeePercentagesProvider.sol

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/vault/IProtocolFeesCollector.sol";
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IProtocolFeePercentagesProvider.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/SingletonAuthentication.sol";
+
+contract ProtocolFeePercentagesProvider is IProtocolFeePercentagesProvider, SingletonAuthentication {
+    IProtocolFeesCollector private immutable _protocolFeesCollector;
+
+    struct FeeTypeData {
+        uint256 value;
+        uint256 maximum;
+        bool registered;
```

Comment:
```suggestion
```

I'd remove this and determine fee type existence from from `feeType.maximum > 0` (We'd need to enforce this on `registerFeeType`). There's not valid reason to create a fee type with a zero maximum value so this doesn't limit us.

Reply:
I added a check for maximum > 0 (as that'd be a mistake), but instead of removing `registered` I packed all three values into a single slot. I prefer that, as we don't need to have too clever logic, nor worry about native fee types (which don't have max or value entries).